netty/src/test/java/io/grpc/netty/UdsNettyChannelProviderTest.java

```diff
@@ -120,6 +121,19 @@ public void managedChannelRegistry_newChannelBuilder() {
     channel.shutdownNow();
   }
 
+  @Test
+  public void managedChannelRegistry_newChannelBuilderForNameResolverRegistry() {
+    Assume.assumeTrue(Utils.isEpollAvailable());
+    ManagedChannelBuilder<?> managedChannelBuilder
+            = Grpc.newChannelBuilderForNameResolverRegistry("unix:///sock.sock",
+             InsecureChannelCredentials.create(), NameResolverRegistry.getDefaultRegistry());
```

Comment:
This isn't testing that the NameResolverRegistry is actually being used. getDefaultRegistry() is already the default.  We need to pass a NameResolverRegistry that clearly behaves differently than the default.

I suggest you create a new registry and add a `io.grpc.internal.testing.FakeNameResolverProvider` to it. The target string and socket address passed to the fake provider don't matter much; you'd just have to pass the same target string to newChannelBuilderForNameResolverRegistry() and the socket address wouldn't be used. If you follow this existing test, then the authority would become "fake-authority."